src/Plugins/Nop.Plugin.Product.Backup/Factory/ProductBackupFactory.cs

```diff
@@ -0,0 +1,146 @@
+
+        if (storeId <= 0)
+            return model;
+
+        model.BackupConfigurationEnabled_OverrideForStore =
+            await _settingService.SettingExistsAsync(productBackupSettings, x => x.BackupConfigurationEnabled, storeId);
+        model.ProcessingProductsNumber_OverrideForStore =
+            await _settingService.SettingExistsAsync(productBackupSettings, x => x.ProcessingProductsNumber, storeId);
+        model.ProductBackupTimer_OverrideForStore =
+            await _settingService.SettingExistsAsync(productBackupSettings, x => x.ProductBackupTimer, storeId);
+        model.ProductBackupStoragePath_OverrideForStore =
+            await _settingService.SettingExistsAsync(productBackupSettings, x => x.ProductBackupStoragePath, storeId);
+
+        return model;
+    }
+
+    public async Task<List<ProductModel>> PrepareProductBackupModel()
+    {
+        if (!_productBackupSettings.BackupConfigurationEnabled) return null;
+
+        var models = await _productService.GetNotExportedProducts();
+        var productModelList = new List<ProductModel>();
+
+        foreach (var model in models)
+        {
+            var pictureModelList = await PrepareImageModel(model.Id);
+
+            var mappedModel = new ProductModel
+            {
+                Id = model.Id,
+                ProductTypeId = model.ProductTypeId,
+                Name = model.Name,
+                ShortDescription = model.ShortDescription,
+                FullDescription = model.ShortDescription,
+                Sku = model.Sku,
+                StockQuantity = model.StockQuantity,
+                OldPrice = model.OldPrice,
+                Price = model.Price,
+                CreatedOnUtc = model.CreatedOnUtc,
+                UpdatedOnUtc = model.UpdatedOnUtc,
+                PictureModelList = pictureModelList
+            };
+            // model.Exported = true;
+            productModelList.Add(mappedModel);
+        }
+
+        return productModelList;
+    }
+
+    public async Task<List<PictureModel>> PrepareImageModel(int id)
+    {
+        var pictureModelList = new List<PictureModel>();
+
+        var pictures = await _pictureService.GetPicturesByProductIdAsync(id);
+
+        foreach (var picture in pictures)
+        {
+            var pictureUrl = await _backupPictureService.GetPictureUrl(picture);
+
+            var pictureModel = new PictureModel()
+            {
+                Id = picture.Id,
+                AltAttribute = picture.AltAttribute,
+                IsNew = picture.IsNew,
+                MimeType = picture.MimeType,
+                SeoFilename = picture.SeoFilename,
+                TitleAttribute = picture.TitleAttribute,
+                VirtualPath = picture.VirtualPath,
+                UrlString = pictureUrl
+            };
+            pictureModelList.Add(pictureModel);
+        }
+
+        return pictureModelList;
+    }
+
+    public async Task ExportModel()
```

Comment:
this is not a factory method, this is a service one, switch this to a dedicated service class